Add EnelGrid integration documentation (#38457)
home-assistant/home-assistant.github.io: source/_integrations/enelgrid.markdown

---
title: EnelGrid
description: Integration for Enel Grid consumption data in Home Assistant

> stage: integration

Reviewer: ```suggestion
```

iot_class: cloud_polling
integration_type: service
dependencies:

>   - recorder

Reviewer: ```suggestion
```

codeowners:
  - "@sathia-musso"
ha_release: 2025.4
ha_domain: enelgrid
ha_integration_type: integration
---

> The **EnelGrid** integration connects Home Assistant with **Enel Italia**, allowing users to import and monitor their electricity consumption data.

Reviewer: ```suggestion
The **EnelGrid** {% term integrations %} connects Home Assistant with **Enel Italia**, allowing users to import and monitor their electricity consumption data.
```
we have a [glossary term reference](https://developers.home-assistant.io/docs/documenting/integration-docs-examples#glossary-term-reference) for this

## Features

- Fetches hourly electricity consumption data
- Tracks daily and cumulative monthly usage
- Integrates with the Home Assistant Energy Dashboard
- Automatically updates daily
- Supports secure re-authentication if needed

> ## Configuration

Reviewer: ```suggestion
## Prerequisites
```

During setup, you’ll be prompted to enter:

- **Username**: Your Enel account email address
- **Password**: Your Enel account password
- **POD Number**: Found on your Enel electricity bill
- **User Number**: Also found on your bill
- **Price per kWh**: Estimated based on your billing rate (total cost ÷ total kWh consumed)

All credentials are stored securely by Home Assistant’s configuration system.

> 

Reviewer: ```suggestion
{% include integrations/config_flow.md %}

```
If the integration supports config flow, there a [reusable text block]({% include integrations/config_flow.md %}) you can use for the Configuration section.

---
title: EnelGrid
description: Integration for Enel Grid consumption data in Home Assistant
stage: integration
iot_class: cloud_polling
integration_type: service
dependencies:
  - recorder
codeowners:
  - "@sathia-musso"
documentation:
  - config_flow
  - has_entity_name
  - entity_unique_id

>   - appropriate_polling

Reviewer: ```suggestion
```
where did you get the documentation category from? What does it do? I don't think this is a component of the header. Please refer to the documentation.
https://github.com/home-assistant/home-assistant.io/pull/38457#issuecomment-2796714312

![Energy Dashboard Configuration screenshot](https://github.com/sathia-musso/enelgrid/raw/main/assets/energy_config.jpg)

If everything is working correctly, your dashboard might look like this:

![Example of Home Assistant Energy Dashboard displaying the EnelGrid sensor](https://github.com/sathia-musso/enelgrid/raw/main/assets/example.jpg)

## 🏷️ Supported Features

| Feature                            | Supported |
|------------------------------------|-----------|
| Hourly Energy Data                 | ✅         |
| Daily Energy Data                  | ✅         |
| Monthly Cumulative Sensor          | ✅         |
| Energy Dashboard Integration       | ✅         |
| Automatic Login                    | ✅         |
| Automatic Daily Fetch              | ✅         |
| Re-authentication Support          | ✅         |

## Additional information

- **Link to related core PR:** [home-assistant/core#142742](https://github.com/home-assistant/core/pull/142742)  
- **Link to parent pull request in the Brands repository:** [home-assistant/brands#6873](https://github.com/home-assistant/brands/pull/6873)  

> - This PR fixes or closes issue: N/A

Reviewer: The information you listed under "Additional information" does not belong in the user documentation; it belongs in the PR template.
Please read the documentation on how to create integration documentation that was provided here:
https://github.com/home-assistant/home-assistant.io/pull/38457#issuecomment-2796714312

## 🔗 External Links

- [Enel Portal](https://www.enel.it/)

> - [Home Assistant Integrations](https://www.home-assistant.io/integrations/)

Reviewer: ```suggestion
```
Links to related topics can be added to the header. Please follow the guideline on this.

## 👤 Author

Developed by [Sathia Francesco Musso](https://github.com/sathia-musso), Contributions welcome!

> 

Reviewer: ```suggestion
```
This information is part of the header, under codeowner.
